sycl/source/builtins/relational_functions.cpp

```diff
@@ -71,17 +71,34 @@ REL_BUILTIN(ONE_ARG, isnormal)
 REL_BUILTIN(TWO_ARGS, isunordered)
 REL_BUILTIN_CUSTOM(TWO_ARGS, isordered,
                    ([](auto x, auto y) { return !sycl::isunordered(x, y); }))
-#if defined(__GNUC__) && !defined(__clang__)
+
+#define GCC_VERSION                                                            \
```

Comment:
I'm not sure if we should use a more distinct name to avoid potential conflicts.

Reply:
Yeah, that might be an idea. Any suggestions?

Looking around the source code, something like `_SYCL_GCC_VER(SION)` or `_SYCL_BUILTINS_GCC_VER(SION)`.

See, e.g., `_PSTL_GCC_VERSION`, `_LIBCPP_GCC_VER`, `LIBC_COMPILER_GCC_VER`, `BENCHMARK_GCC_VERSION`.

Reply:
I've gone with `_SYCL_BUILTINS_GCC_VER`